src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -283,6 +297,31 @@ function compute_precipitation_cache!(
     compute_precipitation_heating!(ᶜSeₜᵖ, ᶜwᵣ, ᶜwₛ, ᶜu, ᶜqᵣ, ᶜqₛ, ᶜts, ᶜ∇T, thp)
 end
 
+function compute_precipitation_surface_fluxes!(
+    Y,
+    p,
+    precip_model::Microphysics1Moment,
+)
+    (; surface_rain_flux, surface_snow_flux) = p.precipitation
+    (; col_integrated_precip_energy_tendency,) = p.conservation_check
+    (; ᶜqᵣ, ᶜqₛ, ᶜwᵣ, ᶜwₛ) = p.precomputed
+
+    (; ᶠtemp_scalar) = p.scratch
+    slg = Fields.level(Fields.local_geometry_field(ᶠtemp_scalar), Fields.half)
+
+    # Constant extrapolation: - put values from bottom cell center to bottom cell face
+    ˢρ = Fields.Field(Fields.field_values(Fields.level(Y.c.ρ, 1)), axes(slg))
+    ˢqᵣ = Fields.Field(Fields.field_values(Fields.level(ᶜqᵣ, 1)), axes(slg))
+    ˢqₛ = Fields.Field(Fields.field_values(Fields.level(ᶜqₛ, 1)), axes(slg))
+    ˢwᵣ = Fields.Field(Fields.field_values(Fields.level(ᶜwᵣ, 1)), axes(slg))
+    ˢwₛ = Fields.Field(Fields.field_values(Fields.level(ᶜwₛ, 1)), axes(slg))
+
+    # Project the flux to C3 vector and convert to physical units.
+    # (This should work with topography)
+    @. surface_rain_flux = projected_vector_data(C3, ˢρ * ˢqᵣ * ˢwᵣ, slg)
```

Comment:
I think we might actually want CT3 instead of C3 here, since CT3 is the one that's orthogonal to topography (and therefore the one that's orthogonal to wind velocity at the surface).

Reply:
I think C3 is orthogonal to topography?

Reply:
Also since we are trying to make it correct with topography...should it be `ˢwᵣ+ w` where w is the physical vertical velocity at the surface (which is not zero when there is topography)?

Reply:
If we use CT3, then we don't need to add any extra component from grid mean velocity because u³ is zero at the surface.

Reply:
CT3 is along the grid, so vertical (no orthogonal)?

Reply:
C3 is orthogonal to wind velocity at the surface in the Euclidean sense (it makes a 90 degree angle), but CT3 is orthogonal in the non-Euclidian sense (the dot product of the unit vector along the CT3 axis with the wind velocity vector is 0 at the surface, based on how we've defined our metric tensor).

Reply:
Yes, sorry, I think orthogonality and u³ being zero are distinct concepts that I got confused. C3 should be orthogonal to CT12, which runs along the surface, in the sense that their dot products are 0.

Reply:
So I should stick with orthogonal to the topography, aka `C3`? And then add `w`. But then, shouldn't I add the whole velocity vector and then do the `C3` projection? Why only add `w`?

Reply:
CT3 might be what we want here because it guarantees that the precipitation flux will be zero when the precipitation velocity is 0. If we use C3, we will always have an additional flux from grid mean velocity (regardless of whether we use w or the full velocity vector) in the presence of topography, which feels wrong to me. 

Sorry about using the word orthogonal; I don't think it's the main thing we need to focus on in this non-Euclidian setting.

Reply:
Oh, wait, is `wᵣ` positive or negative?

Reply:
precipitation_flux is positive upward, in 0M, I think. So there should be a minus sign here if wᵣ is positive downward.

Reply:
Yes, good catch! 

Lets talk tomorrow and figure out w/wo `w` and w/wo `T`